core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/deletevolumes/DeleteVolumesActions.java

```diff
@@ -0,0 +1,229 @@
+import com.sequenceiq.cloudbreak.common.exception.BadRequestException;
+import com.sequenceiq.cloudbreak.converter.spi.ResourceToCloudResourceConverter;
+import com.sequenceiq.cloudbreak.converter.spi.StackToCloudStackConverter;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.AbstractClusterAction;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.ClusterViewContext;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.verticalscale.CoreVerticalScaleEvent;
+import com.sequenceiq.cloudbreak.core.flow2.event.DeleteVolumesTriggerEvent;
+import com.sequenceiq.cloudbreak.core.flow2.stack.AbstractStackFailureAction;
+import com.sequenceiq.cloudbreak.core.flow2.stack.CloudbreakFlowMessageService;
+import com.sequenceiq.cloudbreak.core.flow2.stack.StackFailureContext;
+import com.sequenceiq.cloudbreak.dto.StackDto;
+import com.sequenceiq.cloudbreak.reactor.api.event.StackEvent;
+import com.sequenceiq.cloudbreak.reactor.api.event.StackFailureEvent;
+import com.sequenceiq.cloudbreak.reactor.api.event.resource.DeleteVolumesRequest;
+import com.sequenceiq.cloudbreak.reactor.api.event.resource.DeleteVolumesResult;
+import com.sequenceiq.cloudbreak.service.stack.StackDtoService;
+import com.sequenceiq.cloudbreak.template.model.ServiceComponent;
+import com.sequenceiq.cloudbreak.util.StackUtil;
+import com.sequenceiq.common.api.type.ResourceType;
+
+@Configuration
+public class DeleteVolumesActions {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DeleteVolumesActions.class);
+
+    @Inject
+    private DeleteVolumesService deleteVolumesService;
+
+    @Inject
+    private StackDtoService stackDtoService;
+
+    @Inject
+    private CmTemplateProcessorFactory cmTemplateProcessorFactory;
+
+    @Inject
+    private ResourceToCloudResourceConverter cloudResourceConverter;
+
+    @Inject
+    private CloudbreakFlowMessageService flowMessageService;
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private StackUtil stackUtil;
+
+    @Inject
+    private StackToCloudStackConverter cloudStackConverter;
+
+    @Bean(name = "DELETE_VOLUMES_VALIDATION_STATE")
+    public Action<?, ?> deleteVolumesValidationAction() {
+        return new AbstractClusterAction<>(DeleteVolumesTriggerEvent.class) {
+
+            @Override
+            protected void doExecute(ClusterViewContext context, DeleteVolumesTriggerEvent payload, Map<Object, Object> variables) throws Exception {
+                StackDeleteVolumesRequest stackDeleteVolumesRequest = payload.getStackDeleteVolumesRequest();
+                deleteVolumesService.deleteVolumes(stackDeleteVolumesRequest);
+                StackDto stack = stackDtoService.getById(payload.getResourceId());
+                String requestGroup = stackDeleteVolumesRequest.getGroup();
+                String blueprintText = stack.getBlueprint().getBlueprintText();
+                CmTemplateProcessor processor = cmTemplateProcessorFactory.get(blueprintText);
```

Comment:
Could you please move the business logic into an individual handler? Please only gather the necessary info for request creation and do the necessary status updates in the actions.
For example: https://github.com/hortonworks/cloudbreak/blob/97b3f0cba9a932e958f58f7ae8c46f1fd8995d36/environment/src/main/java/com/sequenceiq/environment/environment/flow/deletion/EnvClustersDeleteActions.java#L103

Reply:
done